diff --git a/src/cmock_internals.h b/src/cmock_internals.h
index 56fb33b..fad67b9 100644
--- a/src/cmock_internals.h
+++ b/src/cmock_internals.h
@@ -79,7 +79,7 @@ extern const char* CMockStringMismatch;
 
 /* amount of memory to allow cmock to use in its internal heap */
 #ifndef CMOCK_MEM_SIZE
-#define CMOCK_MEM_SIZE (32768)
+#define CMOCK_MEM_SIZE (32768*512)
 #endif
 
 /* automatically calculated defs for easier reading */
